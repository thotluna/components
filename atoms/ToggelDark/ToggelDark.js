import React from 'react'
import PropTypes from 'prop-types'
import styles from './ToggelDark.module.css'
import withStyles from '../../hocs/withStyles'

import Icon from '../Icons'

export const ToggelDark = ({ name, isChecked, onChange, getStyles }) => {
  return (
    <div className={getStyles('container-mode')}>
      <input
        className={getStyles('Toggel')}
        aria-label={name}
        type="checkbox"
        checked={isChecked}
        onChange={onChange}
        aria-pressed={isChecked}
        role="button"
      />

      <div className={getStyles('night')}>
        <Icon type="moon" color="inverted" />
      </div>
      <div className={getStyles('day')}>
        <Icon type="sun" color="sun" />
      </div>
    </div>
  )
}

ToggelDark.propTypes = {
  name: PropTypes.string.isRequired,
  getStyles: PropTypes.func.isRequired,
  onChange: PropTypes.func.isRequired,
  isChecked: PropTypes.bool,
}

ToggelDark.defaultProps = {
  isChecked: false,
}

export default withStyles(styles)(ToggelDark)
