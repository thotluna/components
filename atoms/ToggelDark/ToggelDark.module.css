.container-mode {
  position: relative;
  overflow: none;
  height: 35px;
}

.Toggel {
  position: relative;
  width: 70px;
  height: 35px;
  appearance: none;
  background: var(--color-primary-highlight);
  border-radius: 20px;
  box-shadow: inset 0 0 5px var(--color-font-highlight);
  transition: 0.8s;
}

.Toggel:checked {
  background-color: var(--color-primary);
}

.Toggel::before {
  position: absolute;
  top: 0;
  left: 0;
  width: 35px;
  height: 35px;
  background-color: var(--background-color-primary);
  border-radius: 20px;
  box-shadow: 0 2px 5px var(--color-font-highlight);
  content: '';
  transform: scale(1.1);
  transition: 0.4s;
}

.Toggel:checked::before {
  left: 35px;
}

.night,
.day {
  /* height: 35px; */
  position: absolute;
  top: 50%;
  transform: translate(-50%, -50%);
}

.night {
  left: 25%;
}

.day {
  right: 0;
}

.Toggel:not(:checked) ~ .night {
  display: none;
}

.Toggel:checked ~ .night {
  display: block;
}

.Toggel:checked ~ .day {
  display: none;
}

.Toggel:not(:checked) ~ .day {
  display: block;
}
