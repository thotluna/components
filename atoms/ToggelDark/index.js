export { default } from './ToggelDark'
export { options } from './constants'
export { default as styles } from './ToggelDark.module.css'
