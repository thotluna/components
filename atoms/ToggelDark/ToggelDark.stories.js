import React from 'react'
import ToggelDark from '.'

const Template = (args) => <ToggelDark {...args} />

export default {
  title: 'Atoms/ToggelDark',
  component: ToggelDark,
  args: {
    name: 'Dark mode',
  },
  argTypes: {
    onChange: {
      table: {
        disable: true,
      },
    },
  },
}

export const Default = Template.bind({})

export const Active = Template.bind({})
Active.args = { isChecked: true }

export const Inactivo = Template.bind({})
Inactivo.args = { isChecked: false }
