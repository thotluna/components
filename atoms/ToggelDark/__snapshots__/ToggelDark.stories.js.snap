// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Atoms/ToggelDark Active 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "background": "",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
      "padding": "2em",
    }
  }
>
  <div
    className="container-mode"
  >
    <input
      aria-label="Dark mode"
      aria-pressed={true}
      checked={true}
      className="Toggel"
      role="button"
      type="checkbox"
    />
    <div
      className="night"
    >
      <div
        className="container color-inverted size-sm"
      >
        <svg
          fill="none"
          height="100%"
          stroke="currentColor"
          viewBox="0 0 24 24"
          width="100%"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M20.354 15.354A9 9 0 018.646 3.646 9.003 9.003 0 0012 21a9.003 9.003 0 008.354-5.646z"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="1"
          />
        </svg>
      </div>
    </div>
    <div
      className="day"
    >
      <div
        className="container color-sun size-sm"
      >
        <svg
          fill="none"
          height="100%"
          stroke="currentColor"
          viewBox="0 0 24 24"
          width="100%"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 3v1m0 16v1m9-9h-1M4 12H3m15.364 6.364l-.707-.707M6.343 6.343l-.707-.707m12.728 0l-.707.707M6.343 17.657l-.707.707M16 12a4 4 0 11-8 0 4 4 0 018 0z"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth={1}
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`Storyshots Atoms/ToggelDark Default 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "background": "",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
      "padding": "2em",
    }
  }
>
  <div
    className="container-mode"
  >
    <input
      aria-label="Dark mode"
      aria-pressed={false}
      checked={false}
      className="Toggel"
      role="button"
      type="checkbox"
    />
    <div
      className="night"
    >
      <div
        className="container color-inverted size-sm"
      >
        <svg
          fill="none"
          height="100%"
          stroke="currentColor"
          viewBox="0 0 24 24"
          width="100%"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M20.354 15.354A9 9 0 018.646 3.646 9.003 9.003 0 0012 21a9.003 9.003 0 008.354-5.646z"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="1"
          />
        </svg>
      </div>
    </div>
    <div
      className="day"
    >
      <div
        className="container color-sun size-sm"
      >
        <svg
          fill="none"
          height="100%"
          stroke="currentColor"
          viewBox="0 0 24 24"
          width="100%"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 3v1m0 16v1m9-9h-1M4 12H3m15.364 6.364l-.707-.707M6.343 6.343l-.707-.707m12.728 0l-.707.707M6.343 17.657l-.707.707M16 12a4 4 0 11-8 0 4 4 0 018 0z"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth={1}
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;

exports[`Storyshots Atoms/ToggelDark Inactivo 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "background": "",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
      "padding": "2em",
    }
  }
>
  <div
    className="container-mode"
  >
    <input
      aria-label="Dark mode"
      aria-pressed={false}
      checked={false}
      className="Toggel"
      role="button"
      type="checkbox"
    />
    <div
      className="night"
    >
      <div
        className="container color-inverted size-sm"
      >
        <svg
          fill="none"
          height="100%"
          stroke="currentColor"
          viewBox="0 0 24 24"
          width="100%"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M20.354 15.354A9 9 0 018.646 3.646 9.003 9.003 0 0012 21a9.003 9.003 0 008.354-5.646z"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="1"
          />
        </svg>
      </div>
    </div>
    <div
      className="day"
    >
      <div
        className="container color-sun size-sm"
      >
        <svg
          fill="none"
          height="100%"
          stroke="currentColor"
          viewBox="0 0 24 24"
          width="100%"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            d="M12 3v1m0 16v1m9-9h-1M4 12H3m15.364 6.364l-.707-.707M6.343 6.343l-.707-.707m12.728 0l-.707.707M6.343 17.657l-.707.707M16 12a4 4 0 11-8 0 4 4 0 018 0z"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth={1}
          />
        </svg>
      </div>
    </div>
  </div>
</div>
`;
